phys/module_sf_urban.F

```diff
@@ -1133,7 +1133,7 @@ SUBROUTINE urban(LSOLAR,                                           & ! L
           
        CALL TDFCND (DF1,SMR(KZ), QUARTZ, SMCMAX )
        DF1 = DF1 * EXP(-2.0 * SHDFAC)  
-       RGR = EPSV*(RX-SIG*(TGRP**4.)/60.)
+       RGR = EPSV*(RX-SIG*(TA**4.)/60.)
```

Comment:
I think here the original code is correct. We need to use "TGRP" for the green roof surface temperature to compute the outgoing longwave radiation for the greenroof surface not the air. But this RGR does lead to the inconsistence in the "YY" calculation using "TA" a few lines below. I am not quite sure about this yet. this is the reference for this greenroof development: https://link.springer.com/article/10.1007/s10546-014-9991-6

Reply:
@cenlinhe I have tried explaining my part in the attachment here. Please let me know if you need further clarification. Also, please correct me if I am terribly wrong.
[TA_TGRP_Clarification.pdf](https://github.com/wrf-model/WRF/files/15032425/TA_TGRP_Clarification.pdf)

Reply:
@cenlinhe I believe that for consistency in estimating YY, TA needs to be replaced by TGRP while keeping TGRP in the calculations of RGR (as done for the roof case where TRP is used instead), since an iterative approach is implemented. This may provide better results.

Reply:
> @cenlinhe I believe that for consistency in estimating YY, TA needs to be replaced by TGRP while keeping TGRP in the calculations of RGR (as done for the roof case where TRP is used instead), since an iterative approach is implemented. This may provide better results.

I agree that TA should be replaced by TGRP in YY calculation

Reply:
> @cenlinhe I have tried explaining my part in the attachment here. Please let me know if you need further clarification. Also, please correct me if I am terribly wrong. [TA_TGRP_Clarification.pdf](https://github.com/wrf-model/WRF/files/15032425/TA_TGRP_Clarification.pdf)

I agree that current code has some inconsistency in RGR, RR1, and YY calculations. From the greenroof surface energy balance perspective, I think TGRP should be used instead of TA in all these three places. Also, the term (theta0 - T0) is missing but I do not know the reason, likely because the assumption that they are the same.

Reply:
@cenlinhe I still believe that the approximation used in the derivation (\sigma \Theta_s^4 = \sigma \Theta_A^4 (1 + 4(\Theta_s - T_A)/T_A) demands using TA instead of TGRP. By any chance, can we run the simulations without an iterative approach but solve the balance equation with \Theta_A in place of TGRP in the calculations of RGR? It should improve the results.

Reply:
> I think the iterative approach is required to achieve the greenroof surface energy flux balance convergence, which means this equation needs to be inside the iteration loop. Could you please do a test to compare the simulations with TGRP and replacing TGRP by TA (based on your proposed solution) to see how much difference it makes?

@cenlinhe Unfortunately, I do not have computing facilities to run a test simulation. I tried on my local computer but it didn't work well. I will keep you posted if anything changes.

Reply:
Thanks for pointing out the issue. After carefully reviewing the code and the PDF, to be in consistent with the equations, TA shall be used instead of TGRP in the calculation of RGR, RR1, and YY.

For the term  (theta0 - T0), their values shall be close at the first atmospheric level, so I ignore that term during the coding process.

Reply:
> Thanks for pointing out the issue. After carefully reviewing the code and the PDF, to be in consistent with the equations, TA shall be used instead of TGRP in the calculation of RGR, RR1, and YY.
> 
> For the term (theta0 - T0), their values shall be close at the first atmospheric level, so I ignore that term during the coding process.

@jyang104 Thank you so much for your response and for confirming the inconsistency. Now this makes me wonder that when a similar energy balance is applied to a common roof, wall, and ground shouldn't the $\sigma*\theta_s^4$, where $\theta_s$ can represent the surface temperature of roof, wall, and ground, must use a similar approximation $\sigma*\theta_s^4 = \sigma*\theta_A^4 (1 + 4(\theta_s - T_A)/T_A$? I believe that the nonlinear nature of the resulting balance equation in $\theta_s$ is the sole reason for such an approximation. Hence directly using TRP, TWP, or TGP (even if it is from the previous time step) in the calculation of the longwave radiation part of the energy balance causes further inconsistencies. Please correct me if I am wrong. I am extremely new to WRF modeling and trying to understand the physics.

Reply:
@joshi994 You are correct. Part of the reason for the inconsistency is that the calculation of outgoing longwave in the original sf_urban (i.e, for wall and ground and roof) does not involve evapotranspiration that the equation of \sigma*\theta_s^4 is used. When I implemented the green roof, I followed the sf_noahlsm in general that the surface energy balance used the approximation formula. In this case, it will be easier to just correct the green roof part of the code, rather than fixing the ones for TRP, TWP, or TGP.

Reply:
@jyang104 @cenlinhe It make sense corecting the green roof part only. However, it would be nice to see the temperature of the green roof calculated using YY and ZZ1, the linear approximation $\sigma*\theta_s^4 = \sigma*\theta_A^4 (1 + 4(\theta_s - T_A)/T_A$ and using the Newton-Raphson approach right below it. I am leaving it over to the experts to decide and close the request. Thanks for the input @jyang104 and @cenlinhe for providing me the opportunity to speak with @jyang104.